src/WpfMath/UserDefinedColorParser.cs

```diff
@@ -0,0 +1,264 @@
+    /// </summary>
+    public class UserDefinedColorParser
+    {
+        /// <summary>
+        /// Parses the <paramref name="input"/> to its <see cref="Color"/> equivalent.
+        /// </summary>
+        /// <param name="input"></param>
+        /// <returns></returns>
+        public static Color ParseUserColor(string input)
+        {
+            Color resultColor = new Color();
+            ColorStringTypes CST = GetColorString(input.Trim());
+            switch (CST)
+            {
+                case ColorStringTypes.Byte_longString:
+                    {
+                        byte[] channelbytes = ByteStringValues(input.Trim(), 4);
+                        resultColor.A = channelbytes[0];
+                        resultColor.R = channelbytes[1];
+                        resultColor.G = channelbytes[2];
+                        resultColor.B = channelbytes[3];
+                        break;
+                    }
+                    
+                case ColorStringTypes.Byte_shortString:
+                    {
+                        byte[] channelbytes = ByteStringValues(input.Trim(), 3);
+                        resultColor.A = 255;
+                        resultColor.R = channelbytes[0];
+                        resultColor.G = channelbytes[1];
+                        resultColor.B = channelbytes[2];
+                        break;
+                    }
+                case ColorStringTypes.Hex_longString:
+                    {
+                        byte[] channelbytes = ByteHexStringValues(input.Trim(), 8);
+                        resultColor.A = channelbytes[0];
+                        resultColor.R = channelbytes[1];
+                        resultColor.G = channelbytes[2];
+                        resultColor.B = channelbytes[3];
+                        break;
+                    }
+                case ColorStringTypes.Hex_shortString:
+                    {
+                        byte[] channelbytes = ByteHexStringValues(input.Trim(), 6);
+                        resultColor.A = 255;
+                        resultColor.R = channelbytes[0];
+                        resultColor.G = channelbytes[1];
+                        resultColor.B = channelbytes[2];
+                        break;
+                    }
+                case ColorStringTypes.None:
+                    throw new TexParseException($"The text: {input} could not be parsed.");
+                default:
+                    break;
+            }
+            return resultColor;
+        }
+
+        #region Color Parsing Helpers
+        /// <summary>
+        /// Specifies the type of <see cref="Color"/> the string might represent.
+        /// </summary>
+        private enum ColorStringTypes
+        {
+            /// <summary>
+            /// Type is a <see cref="string"/> consisting of byte values of the color.
+            /// </summary>
+            Byte_longString,
```

Comment:
Could you please fix the naming here? Underscore in an enum item name seems odd. Please make that `ByteLongString` etc.

Also I recommend to rename these to something like `CommaSeparatedDecimal`, `PrefixedHex` etc.